chain/chain/src/flat_storage_resharder.rs

```diff
@@ -1457,6 +1466,35 @@ mod tests {
         )
     }
 
+    /// Shard layout with three shards.
+    fn shards_layout_three_shards() -> ShardLayout {
+        let s0 = ShardId::new(0);
+        let s1 = ShardId::new(1);
+        let s2 = ShardId::new(2);
+        let shards_split_map = BTreeMap::from([(s0, vec![s0]), (s1, vec![s1]), (s2, vec![s2])]);
+        ShardLayout::v2(
+            vec![account!("ff"), account!("pp")],
+            vec![s0, s1, s2],
+            Some(shards_split_map),
+        )
+    }
+
+    /// Derived from [shards_layout_three_shards] by splitting the second shard.
+    fn shards_layout_three_shards_after_split() -> ShardLayout {
+        let s0 = ShardId::new(0);
+        let s1 = ShardId::new(1);
+        let s2 = ShardId::new(2);
+        let s3 = ShardId::new(3);
+        let s4 = ShardId::new(4);
+
+        let shards_split_map = BTreeMap::from([(s0, vec![s0]), (s1, vec![s3, s4]), (s2, vec![s2])]);
+        ShardLayout::v2(
+            vec![account!("ff"), account!("pp"), account!("yy")],
+            vec![s0, s3, s4, s2],
+            Some(shards_split_map),
+        )
+    }
```

Comment:
There is a method to derive a new shard layout from old one by splitting a shard - please grep for derive in shard_layout.rs